src/lib.rs

```diff
@@ -67,202 +77,130 @@ world.spawn((
-#[action_output(bool)]
-struct Jump;
-
-#[derive(InputAction)]
-#[action_output(bool)]
-struct Fire;
 ```
 
 By default, input is read from all connected gamepads. You can customize this by adding the [`GamepadDevice`] component to the
 context entity.
 
-Context actions will be evaluated in the schedule associated at context registration. Contexts registered in the same
-schedule will be evaluated in their spawning order, but you can override it by adding the [`ContextPriority`] component.
-You can also activate or deactivate contexts by inserting [`ContextActivity`] component.
+## Contexts
 
-Actions also have [`ActionSettings`] component that customizes their behavior.
+Contexts define when actions are evaluated. They are associated with action entities via the [`Actions<C>`] relationship,
+where `C` is your context type, and can be quickly bound to them using the [`actions!`] macro.
 
-## Input modifiers
+Inside of this crate, contexts are stored using regular components, commonly on an entity for which the input is associated (player, button, dialog, etc.).
+Depending on your type of game, you may have a single global context
+or multiple contexts for different gameplay states. For games with multiple entities driven by a single context it's
+common to create a "controller" entity which applies the input to the desired entity.
 
-Action values are stored in two forms:
-- In a typed form, as the [`Action<C>`] component.
-- In a dynamically typed form, as the [`ActionValue`], which is one of the required components of [`Action<C>`].
-  Its variant depends on the [`InputAction::Output`].
+Contexts can be activated or deactivated using the [`ContextActivity`] component.
+By default, contexts are active when the component is present.
+When active, all actions associated with the context are evaluated.
 
-During [`EnhancedInputSet::Update`], input is read for each [`Binding`] as an [`ActionValue`], with the variant depending
-on the input source. This value is then converted into the [`ActionValue`] on the associated action entity. For example,
-key inputs are captured as [`bool`], but if the action's output type is [`Vec2`], the value will be assigned to the X axis
-as `0.0` or `1.0`. See [`Binding`] for details on how each source is captured, and [`ActionValue::convert`] for how values
-are transformed.
+Contexts can be [layered](crate::context::ContextPriority), causing one context to rely on another's presence, and any number of them can be active at the same time.
+To register a component as an input context, you need to call [`InputContextAppExt::add_input_context`]. By default, contexts are
+evaluated during [`PreUpdate`], but you can customize this by using [`InputContextAppExt::add_input_context_to`] instead.
 
-Then, during [`EnhancedInputSet::Apply`], the value from [`ActionValue`] is written into [`Action<C>`].
+Context actions will be evaluated in the schedule associated at context registration. Contexts registered in the same
```

Comment:
Looks like this paragraph duplicates some of the information from the previous two.